optimum/habana/transformers/models/qwen2/modeling_qwen2.py

```diff
@@ -1149,3 +1154,221 @@ def apply_customized_rope(q, k, cos, sin, position_ids, training=True):
     else:
         # keep the same implementation as Transformers v4.37.2
         return apply_rotary_pos_emb(q, k, cos[position_ids], sin[position_ids])
+
+class GaudiQwen2ForSequenceClassification(Qwen2ForSequenceClassification):
+    def __init__(self, config):
+        super().__init__(config)
+        self.num_labels = config.num_labels
+        self.model = GaudiQwen2Model(config)
```

Comment:
my point is that the line in adapt_transformers_to_gaudi: `transformers.models.qwen2.modeling_qwen2.Qwen2Model = GaudiQwen2Model` already makes these qwen2 models with task heads compatible/optimized for gaudi.